src/systems/camera-system.js

```diff
@@ -205,18 +205,41 @@ function getAudio(o) {
 
 const FALLOFF = 0.9;
 export class CameraSystem {
-  constructor(scene) {
+  constructor(camera, renderer) {
+    this.viewingCamera = camera;
     this.lightsEnabled = localStorage.getItem("show-background-while-inspecting") === "true";
     this.verticalDelta = 0;
     this.horizontalDelta = 0;
     this.inspectZoom = 0;
     this.mode = CAMERA_MODE_SCENE_PREVIEW;
     this.snapshot = { audioTransform: new THREE.Matrix4(), matrixWorld: new THREE.Matrix4() };
     this.audioSourceTargetTransform = new THREE.Matrix4();
+
+    if (customFOV) {
+      this.viewingCamera.fov = customFOV;
+    }
+    this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_VIDEO_TEXTURE_TARGET);
+    this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_FIRST_PERSON_ONLY);
+
+    // xr.updateCamera gets called every render to copy the active cameras properties to the XR cameras. We also want to copy layers.
+    // TODO this logic should either be moved into THREE or removed when we ditch aframe camera system
+    const xrManager = renderer.xr;
+    const updateXRCamera = xrManager.updateCamera;
+    xrManager.updateCamera = function(camera) {
+      updateXRCamera(camera);
+      const xrCamera = xrManager.getCamera();
+      xrCamera.layers.mask = camera.layers.mask;
+      if (xrCamera.cameras.length) {
```

Comment:
Is it guaranteed that length equals to or is greater than 2 onless it is zero? If so, I think we need comment here for future readers.

Reply:
Yeah this is internal state of XRManager https://github.com/MozillaReality/three.js/blob/hubs/master/src/renderers/webvr/WebXRManager.js#L47

This code should actually be applied as a threejs patch but I didn't want an additional thing to juggle in this PR